lib/src/widgets/custom_credit_card.dart

```diff
@@ -0,0 +1,209 @@
+                      ],
+                      validator: (String? input) =>
+                          CardUtils.validateCVC(input, widget.locale),
+                      onSaved: (value) {
+                        _cardData.cvc = value ?? '';
+                        widget.onCreditCardFormChange(
+                            _cardData, _isValidForm());
+                      }),
+                ),
+              ],
+            )
+          else ...[
+            CardFormField(
+              inputDecoration: buildInputDecoration(
+                hintText: '${widget.locale.expiry} (MM / YY)',
+              ),
+              inputFormatters: [
+                FilteringTextInputFormatter.digitsOnly,
+                LengthLimitingTextInputFormatter(4),
+                CardMonthInputFormatter(),
+              ],
+              validator: (String? input) =>
+                  CardUtils.validateDate(input, widget.locale),
+              onSaved: (value) {
+                List<String> expireDate = CardUtils.getExpiryDate(value!);
+                if (expireDate.length == 2) {
+                  _cardData.month = expireDate.first;
+                  _cardData.year = expireDate[1];
+                }
+                widget.onCreditCardFormChange(_cardData, _isValidForm());
+              },
+            ),
+            CardFormField(
+                inputDecoration: buildInputDecoration(
+                  hintText: widget.locale.cvc,
+                ),
+                inputFormatters: [
+                  FilteringTextInputFormatter.digitsOnly,
+                  LengthLimitingTextInputFormatter(4),
+                ],
+                validator: (String? input) =>
+                    CardUtils.validateCVC(input, widget.locale),
+                onSaved: (value) {
+                  _cardData.cvc = value ?? '';
+                  widget.onCreditCardFormChange(_cardData, _isValidForm());
+                }),
+          ],
+        ],
+      ),
+    );
+  }
+}
+
+InputDecoration buildInputDecoration(
+    {required String hintText, bool addNetworkIcons = false}) {
+  return InputDecoration(
+      suffixIcon: addNetworkIcons ? const NetworkIcons() : null,
+      hintText: hintText,
+      focusedErrorBorder: defaultErrorBorder,
+      enabledBorder: defaultEnabledBorder,
+      focusedBorder: defaultFocusedBorder,
+      errorBorder: defaultErrorBorder);
+}
+
+
+BorderRadius defaultBorderRadius = const BorderRadius.all(Radius.circular(8));
+
+OutlineInputBorder defaultEnabledBorder = OutlineInputBorder(
+    borderSide: BorderSide(color: Colors.grey[400]!),
+    borderRadius: defaultBorderRadius);
+
+OutlineInputBorder defaultFocusedBorder = OutlineInputBorder(
+    borderSide: BorderSide(color: Colors.grey[600]!),
+    borderRadius: defaultBorderRadius);
+
+OutlineInputBorder defaultErrorBorder = OutlineInputBorder(
+    borderSide: const BorderSide(color: Colors.red),
+    borderRadius: defaultBorderRadius);
```

Comment:
1. few commas missing
2. shoud be inside `_CustomCreditCardState`
3. should be private